perf/perf_24x7_all_events.py

```diff
@@ -0,0 +1,135 @@
+        self.event_sysfs = "/sys/bus/event_source/devices/hv_24x7"
+
+        # Check if this is a guest
+        # 24x7 is not suported on guest
+        cpu_output = process.run("cat /proc/cpuinfo")
+        if "emulated by" in cpu_output.stdout:
+            self.cancel("This test is not supported on guest")
+
+        # Check if 24x7 is present
+        if os.path.exists(self.event_sysfs):
+            self.log.info('hv_24x7 present')
+        else:
+            self.cancel("%s doesn't exist.This feature is supported"
+                        " only on LPAR" % self.event_sysfs)
+
+        # Performance measurement has to be enabled in lpar through BMC
+        # Check if its enabled
+        result_perf = process.run("%s,domain=2,core=1/ sleep 1"
+                                  % self.perf_stat, ignore_status=True)
+        if "not supported" in result_perf.stderr:
+            self.cancel("Please enable LPAR to allow collecting"
+                        " the 24x7 counters info")
+
+        # Getting the number of cores
+        output = process.run("lscpu")
+        for line in output.stdout.split('\n'):
+            if 'Core(s) per socket:' in line:
+                self.cores = int(line.split(':')[1].strip())
+
+        # Getting the number of chips available in the machine
+        self.chip = memory.numa_nodes()
+
+        # Collect all hv_24x7 events
+        self.list_of_hv_24x7_events = []
+        for line in process.get_perf_events('hv_24x7'):
+            line = line.split(',')[0].split('/')[1]
+            self.list_of_hv_24x7_events.append(line)
+
+        # Clear the dmesg, by that we can capture the delta at the end of the test.
+        process.run("dmesg -c")
+
+    def test_all_events(self):
+        perf_args = "-C 9 -v -e"
+        for line in self.list_of_hv_24x7_events:
+            if line.startswith('HP') or line.startswith('CP'):
+                # Running for domain range from 1-6
+                for domain in range(1, 7):
+                    for core in range(0, self.cores + 1):
+                        events = "hv_24x7/%s,domain=%s,core=%s/" % (line, domain, core)
+                        cmd = 'perf stat %s %s sleep 1' % (perf_args, events)
+                        if process.system(cmd, ignore_status=True):
+                            self.fail_cmd.append(cmd)
+            else:
+                for chip_item in self.chip:
+                    events = "hv_24x7/%s,chip=%s/" % (line, chip_item)
+                    cmd = "perf stat %s %s sleep 1" % (perf_args, events)
+                    if process.system(cmd, ignore_status=True):
+                        self.fail_cmd.append(cmd)
+
+        if len(self.fail_cmd) > 0:
+            for cmd in range(len(self.fail_cmd)):
+                self.log.info("Failed command: %s" % self.fail_cmd[cmd])
+            self.fail("hv_24x7: some of the commands failed, refer to log")
+
+    def tearDown(self):
+        # Collect the dmesg
+        process.run("dmesg -T")
```

Comment:
are we doing any verification after collecting dmesg log ?

Reply:
Just logging. Very rarely errors comes from this, in that case just wanted to log with out missing.

Reply:
ah ok sure